Mathlib/Tactic/FieldSimp.lean

```diff
@@ -151,7 +151,7 @@ syntax (name := fieldSimp) "field_simp" (config)? (discharger)? (&" only")?
 elab_rules : tactic
 | `(tactic| field_simp $[$cfg:config]? $[(discharger := $dis)]? $[only%$only?]?
     $[$sa:simpArgs]? $[$loc:location]?) => withMainContext do
-  let cfg ← elabSimpConfig (cfg.getD ⟨.missing⟩) .simp
+  let cfg ← elabSimpConfig (mkOptionalNode cfg) .simp
```

Comment:
You should also be able to move the optional node into `cfg` by writing something like `$[$cfg:config ?]` rather than `$[$cfg:config]?`. That's how `norm_num` does it at least.

Reply:
Like this? https://github.com/leanprover-community/mathlib4/blob/86b7828a1e4debe4984e5051220a6026c8d4c6a1/Mathlib/Tactic/NormNum/Core.lean#L318-L320

That's using `elab`, which adds an extra layer of macro expansion on top of `elab_rules`, and apparently uses a totally different matching syntax. I was unable to find a way to get my `elab_rules` match to behave in that way. (And I prefer not to switch over to the `elab` sytnax, because I understand it less.)

Reply:
You're right, `elab_rules` seems to work quite differently. It looks like `Tauto`, `Congr!` and `LiftLets` all follow the same `mkOptionalNode` pattern as here. So this is probably best.